docs/features/nullable-reference-types.md

```diff
@@ -157,6 +157,9 @@ var y = (IEnumerable<object>)x;  // warning
 var z = (IEnumerable<object?>)x; // no warning
 ```
 
+### Method type inference
+_Describe details_
```

Comment:
If possible, I'd suggest filing this in as part of this PR. That helps with review and also it's good to do it while fresh in your mind. Thanks